pkg/client/cli/cmd_gatherlogs.go

```diff
@@ -260,12 +317,122 @@ func zipFiles(files []string, zipFileName string) error {
+// the anonymizer.
+func getPodName(podName string, anon bool, anonymizer *anonymizer) string {
+	// If we aren't anonymizing the logs, just return the podName
+	if !anon {
+		return podName
+	}
+
+	// If this pod name has already been mapped, return that
+	if anonName, ok := anonymizer.podNames[podName]; ok {
+		return anonName
+	}
+
+	// the podName hasn't been anonymized yet so we split it up
+	// so we can anonymize the namespace
+	nameComponents := strings.SplitN(podName, ".", 2)
+	if len(nameComponents) != 2 {
+		unknownPodName := "anonPod.anonNamespace"
+		anonymizer.podNames[podName] = unknownPodName
+		return unknownPodName
+	}
+	var anonPodName, anonNamespace string
+	name, namespace := nameComponents[0], nameComponents[1]
+	if val, ok := anonymizer.namespaces[namespace]; ok {
+		anonNamespace = val
+	} else {
+		anonNamespace = fmt.Sprintf("namespace-%d", len(anonymizer.namespaces)+1)
+		anonymizer.namespaces[namespace] = anonNamespace
+	}
+
+	// we want to special case the traffic-manager so we can easily distinguish
+	// between that and the traffic-agents
+	if strings.Contains(name, "traffic-manager") {
+		anonPodName = fmt.Sprintf("traffic-manager.%s", anonNamespace)
+	} else {
+		anonPodName = fmt.Sprintf("pod-%d.%s", len(anonymizer.podNames)+1, anonNamespace)
+	}
+	// Store the anonPodName in the map
+	anonymizer.podNames[podName] = anonPodName
+	return anonPodName
+}
+
+// anonymizeLog is a helper function that replaces the namespace + podName
+// used in the log with its anonymized version, provided by the anonymizer.
+// It overwrites the file with the anonymized version.
+func anonymizeLog(stdout io.Writer, logFile string, anonymizer *anonymizer) error {
+	// Read the contents we are going to overwrite from the file
+	content, err := os.ReadFile(logFile)
+	if err != nil {
+		return err
+	}
+	// Open the file with write so we can overwrite it
+	stringContent := string(content)
+	f, err := os.OpenFile(logFile, os.O_RDWR, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	// First we replace the actual namespace with the anonymized
+	// version.
+	for namespace, anonNamespace := range anonymizer.namespaces {
+		stringContent = strings.ReplaceAll(stringContent, namespace, anonNamespace)
+	}
+	// Now we do pod name which is a little bit more complicated
+	for fullPodName, fullAnonPodName := range anonymizer.podNames {
+		// strip the namespace off of the anonymized name
+		anonPodParts := strings.Split(fullAnonPodName, ".")
+		anonPodName := anonPodParts[0]
+
+		// Strip the namespace off of the podName
+		podParts := strings.Split(fullPodName, ".")
+
+		// replace pod name + full hash with the anonymized name
+		podName := podParts[0]
+		stringContent = strings.ReplaceAll(stringContent, podName, anonPodName)
+
+		// Strip the hash off of the pod name + replace remainder with anonymized name
+		brokenPodName := strings.Split(podName, "-")
+		podName = strings.Join(brokenPodName[:len(brokenPodName)-2], "-")
```

Comment:
watch out cause this is not going to work for statefulsets -- if a statefulset is called `my-sts` the pods will be `my-sts-0`, `my-sts-1`, etc -- you might need a regex of some sort to check what kind of pod it is

this will similarly have trouble for pods created from Jobs which also have their own format IIRC

Reply:
ah yeah let me think about that... I don't think we need to worry about Jobs (yet) since we can't intercept those (afaik), but it won't hurt to be forward thinking if it's not too difficult.